test/e2e/adminapi_redeployvm.go

```diff
@@ -38,63 +38,18 @@ var _ = Describe("[Admin API] VM redeploy action", func() {
 		vm := vms[0]
 		log.Infof("selected vm: %s", *vm.Name)
 
-		By("triggering the redeploy action")
-		clockDrift := -1 * time.Minute
-		startTime := time.Now().Add(clockDrift)
+		By("verifying redeploy action completes without error")
 		resp, err := adminRequest(ctx, http.MethodPost, "/admin"+resourceID+"/redeployvm", url.Values{"vmName": []string{*vm.Name}}, nil, nil)
 		Expect(err).NotTo(HaveOccurred())
 		Expect(resp.StatusCode).To(Equal(http.StatusOK))
 
-		By("verifying through cluster events that the redeployment happened")
-		err = wait.PollImmediate(10*time.Second, 10*time.Minute, func() (bool, error) {
-			events, err := clients.Kubernetes.EventsV1().Events("default").List(ctx, metav1.ListOptions{})
-
-			if err != nil {
-				return false, err
-			}
-
-			var nodeKillTime metav1.MicroTime
-
-			for _, event := range events.Items {
-				if nodeKillTime.IsZero() &&
-					event.Reason == "TerminationStart" &&
-					!event.CreationTimestamp.After(startTime) {
-					nodeKillTime = metav1.MicroTime(event.CreationTimestamp)
-					break
-				}
-
-			}
-
-			var nodeNotReady, rebooted, nodeReady bool
-
-			for _, event := range events.Items {
-				if event.CreationTimestamp.After(nodeKillTime.Time) && event.Regarding.Name == *vm.Name {
-					if !nodeNotReady && event.Reason == "NodeNotReady" {
-						nodeNotReady = true
-						log.Info("node entered not ready state")
-					}
-					if !rebooted && event.Reason == "Rebooted" {
-						rebooted = true
-						log.Info("node reboooted")
-					}
-					if !nodeReady && event.Reason == "NodeReady" {
-						nodeReady = true
-						log.Info("node returned to ready state")
-						break
-					}
-				}
-			}
-
-			return nodeNotReady && rebooted && nodeReady, nil
-		})
-		Expect(err).NotTo(HaveOccurred())
-
-		By("waiting for all nodes to be ready")
-		err = wait.PollImmediate(10*time.Second, 10*time.Minute, func() (bool, error) {
+		By("verifying all nodes are eventually ready")
+		// wait 1 minute - this will guarantee we pass the minimum (default) threshold of Node heartbeats (40 seconds)
+		err = wait.Poll(1*time.Minute, 10*time.Minute, func() (bool, error) {
```

Comment:
Can we wait on / poll only the single node we redeployed, rather than all nodes, for readiness? My concern is that this new test could flake if another node was NotReady for an unrelated reason, since we're now only asserting readiness.

Reply:
Agree, lets just wait for the vm that we redeployed to return to ready. 

An additional concern is a false pass here; if within 60 seconds the test passes with "Node Ready" before the azure infrastructure layer even has the time to stop, relocate, and start the VM.

Reply:
@bryanro92 - By the time we've reached this code, we've already guaranteed 1.) a queue of a "redeploy" action on the VM, followed by 2.) a "wait until all is well" - I can add a unit test to cover that case.

re: https://github.com/Azure/ARO-RP/blob/85fb1cc131eefaeadda68f6dded7bc6cdd1be5a2/pkg/util/azureclient/mgmt/compute/virtualmachines_addons.go#L39-L46

Reply:
nice, that addresses my second concern, thanks @SudoBrendan. Lets just verify the vm we redeployed returns to healthy state (vs list) and call this flake fixed  :crossed_fingers: